### PR TITLE
hibernate-relations-hw solution

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/Main.java [hunk 1]
```diff
@@ -16,17 +19,17 @@ public static void main(String[] args) {
         SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
 
         Country usa = new Country("USA");
-        CountryService countryService = null; // TODO: initialize this instance
+        CountryService countryService = new CountryServiceImpl(sessionFactory);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/Main.java:22]:
```suggestion
        CountryService countryService = new CountryServiceImpl(new CountryDaoImpl(sessionFactory));
```

same for other

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java [hunk 2]
```diff
@@ -12,11 +15,36 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
...
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        Session session = factory.openSession();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:42]: use try wth resources + catch

File: src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -12,11 +15,36 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
...
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Session session = factory.openSession();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java:42]: same

File: src/main/java/mate/academy/hibernate/relations/model/Actor.java [hunk 4]
```diff
@@ -1,8 +1,20 @@
...
 
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "Actor")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Actor.java:11]:
```suggestion
@Table(name = "actors")
```

read table naming convention

Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Actor.java:11]:
```suggestion
@Table(name = "actors")
```

read table naming convention

File: src/main/java/mate/academy/hibernate/relations/model/Actor.java [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
 public class Actor implements Cloneable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String name;
+    @ManyToOne()
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Actor.java:17]:
```suggestion
    @ManyToOne
```

File: src/main/java/mate/academy/hibernate/relations/model/Country.java [hunk 6]
```diff
@@ -1,6 +1,16 @@
 package mate.academy.hibernate.relations.model;
 
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "Country")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Country.java:10]: rename

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 7]
```diff
@@ -1,11 +1,23 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.Table;
 import java.util.ArrayList;
 import java.util.List;
 
+@Entity
+@Table(name = "Movie")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:13]: rename

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 8]
```diff
@@ -1,11 +1,23 @@
...
 public class Movie implements Cloneable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String title;
+
+    @ManyToMany
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:20]: Add `@JoinTable`

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java [hunk 9]
```diff
@@ -1,16 +1,26 @@
...
 import mate.academy.hibernate.relations.service.ActorService;
+import mate.academy.hibernate.relations.util.HibernateUtil;
+import org.hibernate.SessionFactory;
 
 public class ActorServiceImpl implements ActorService {
+    private final ActorDao actorDao;
+
+    public ActorServiceImpl(SessionFactory sessionFactory) {
+        actorDao = new ActorDaoImpl(HibernateUtil.getSessionFactory());
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:15]:
```suggestion
    public ActorServiceImpl(ActorDao actorDao) {
        this.actorDao = actorDao;
    }
```

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java [hunk 10]
```diff
@@ -1,16 +1,26 @@
...
     @Override
     public Actor add(Actor actor) {
-        return null;
+        return actorDao.add(actor);
     }
 
     @Override
     public Actor get(Long id) {
-        return null;
+        return actorDao.get(id).orElse(null);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:24]: orElse(null) - bad approach, use orElseThrow

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 11]
```diff
@@ -1,16 +1,25 @@
...
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:12]: same

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 12]
```diff
@@ -1,16 +1,25 @@
...
     @Override
     public Country add(Country country) {
-        return null;
+        return countryDao.add(country);
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        return countryDao.get(id).orElse(null);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:23]: same

File: src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java [hunk 13]
```diff
@@ -1,16 +1,25 @@
...
+import mate.academy.hibernate.relations.dao.MovieDao;
+import mate.academy.hibernate.relations.dao.impl.MovieDaoImpl;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import org.hibernate.SessionFactory;
 
 public class MovieServiceImpl implements MovieService {
+    private final MovieDao movieDao;
+
+    public MovieServiceImpl(SessionFactory sessionFactory) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java:12]: same

File: src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java [hunk 14]
```diff
@@ -1,16 +1,25 @@
...
     @Override
     public Movie add(Movie movie) {
-        return null;
+        return movieDao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+        return movieDao.get(id).orElse(null);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java:23]: same

File: src/main/resources/hibernate.cfg.xml [hunk 15]
```diff
@@ -0,0 +1,15 @@
+<hibernate-configuration>
+    <session-factory>
+        <property name="hibernate.dialect">org.hibernate.dialect.MySQL8Dialect</property>
+        <property name="connection.url">jdbc:mysql://localhost:3306/test</property>
+        <property name="connection.driver_class">com.mysql.cj.jdbc.Driver</property>
+        <property name="connection.username">root</property>
+        <property name="connection.password">henghfdf</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create</property>
```
Comment (Elena-Bruyako) [inline src/main/resources/hibernate.cfg.xml:9]:
```suggestion
        <property name="hbm2ddl.auto">create-drop</property>
```

File: pom.xml [hunk 16]
```diff
@@ -19,6 +19,21 @@
...
     <dependencies>
+        <!-- https://mvnrepository.com/artifact/org.hibernate.orm/hibernate-core -->
+        <dependency>
+            <groupId>org.hibernate.orm</groupId>
+            <artifactId>hibernate-core</artifactId>
+            <version>6.5.2.Final</version>
+        </dependency>
+
+
+        <!-- https://mvnrepository.com/artifact/com.mysql/mysql-connector-j -->
```
Comment (Elena-Bruyako) [inline pom.xml:30]:
```suggestion
```

File: src/main/java/mate/academy/hibernate/relations/model/Actor.java [hunk 17]
```diff
@@ -1,8 +1,20 @@
...
 
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "actor")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Actor.java:11]:
```suggestion
@Table(name = "actors")
```

File: src/main/java/mate/academy/hibernate/relations/model/Actor.java (hunk 5 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/hibernate/relations/model/Actor.java:17]: not fixed

File: src/main/java/mate/academy/hibernate/relations/model/Country.java [hunk 18]
```diff
@@ -1,6 +1,16 @@
 package mate.academy.hibernate.relations.model;
 
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "country")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Country.java:10]:
```suggestion
@Table(name = "countries")
```

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 19]
```diff
@@ -1,11 +1,23 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.Table;
 import java.util.ArrayList;
 import java.util.List;
 
+@Entity
+@Table(name = "movie")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:13]:
```suggestion
@Table(name = "movies")
```

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java (hunk 8 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/hibernate/relations/model/Movie.java:20]: not fixed

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java (hunk 10 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:21]: not fixed

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java (hunk 12 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:21]: not fixed

File: src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java (hunk 14 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java:21]: not fixed

Comment (mkazarian-1) [reply src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java:21]: these comments were hidden so I didn't see them, 
sorry

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request closed without merging**